src/Raven.Server/Smuggler/Documents/Iteration/IterationState.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils.Enumerators;
+using Sparrow;
+using Voron.Impl;
+
+namespace Raven.Server.Smuggler.Documents.Iteration
+{
+    public abstract class IterationState<T> : PulsedEnumerationState<T>
+    {
+        protected IterationState(DocumentsOperationContext context)
+        {
+            Context = context;
+        }
+
+        protected readonly Size PulseLimit = new Size(16, SizeUnit.Megabytes); // TODO arek - make it configurable
```

Comment:
Yes, making this configurable is highly desired.
For that matter, should we allow it to be enabled in 64 bits as well?
Might be relevant for 64 bits systems using encrypted dbs,  no?

Reply:
We need this if running in 32bits or using encryption, otherwise `GetTotal32BitsMappedSize()` and `TotalEncryptionBufferSize` are always `0` so it doesn't make sense to do any checks.

Not sure if 16MB limit should be allowed to be configurable for a user.

Reply:
Yes, we want to allow this to be configuration